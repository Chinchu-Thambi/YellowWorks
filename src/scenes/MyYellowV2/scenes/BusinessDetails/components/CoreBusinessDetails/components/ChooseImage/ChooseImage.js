import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../../../../../../../../components/Modal';
import { EditIconButton } from '../../../../../../components/IconButtons';
import AssetsManagement from '../../../../../../components/media/AssetsManagement';
import uploadImageIcon from '../../../../../../../../assets/icons/image-upload-icon.svg';

const ChooseImage = ({
  id, label, value, onSelect, circle = false, small = false,
}) => {
  const [isModalVisible, setIsModalVisible] = React.useState(false);
  const [currentValue, setCurrentValue] = React.useState(value);

  const toggleModal = () => {
    setIsModalVisible(!isModalVisible);
  };

  const handleSelect = (contentUrl) => {
    onSelect(contentUrl);
    setCurrentValue(contentUrl);
    toggleModal();
  };

  const getDimensions = () => {
    if (circle) {
      return 'h-10 w-10';
    }
    if (currentValue) {
      return 'max-h-10 max-w-10';
    }
    return 'h-7 w-10';
  };

  return (
    <>
      <label htmlFor={id}>{label}</label>

      <div className="flex items-center space-x-3">
        <button id={id} className={`bg-transparent ${getDimensions()}`} type="button" onClick={toggleModal}>
          {currentValue && (
            <img
              className={`w-full h-full object-cover ${circle ? 'rounded-full' : 'rounded-md'} ${small && 'max-h-10'}`}
              src={currentValue}
              alt={label}
            />
          )}
          {!currentValue && (
            <div
              className={`bg-violet-100 text-violet-400 border border-dashed border-violet-300 flex items-center justify-center w-full h-full object-cover ${
                circle ? 'rounded-full' : 'rounded-md'
              }`}
            >
              <img src={uploadImageIcon} className="h-4" alt="add" />
            </div>
          )}
        </button>

        <EditIconButton onClick={toggleModal} label="Change" />
      </div>

      <Modal title={`Choose an image for ${label}`} isVisible={isModalVisible} onDismiss={toggleModal}>
        <AssetsManagement type="ImageObject" onSelect={handleSelect} />
      </Modal>
    </>
  );
};

ChooseImage.defaultProps = {
  id: 'choose-image',
  value: '',
  label: '',
  onSelect: () => {},
  circle: false,
  small: false,
};

ChooseImage.propTypes = {
  id: PropTypes.string,
  value: PropTypes.string,
  label: PropTypes.string,
  onSelect: PropTypes.func,
  circle: PropTypes.bool,
  small: PropTypes.bool,
};

export default ChooseImage;
