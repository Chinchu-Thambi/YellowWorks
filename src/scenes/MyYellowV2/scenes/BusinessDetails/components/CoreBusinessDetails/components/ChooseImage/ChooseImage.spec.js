import React from 'react';
import { render, screen } from '@testing-library/react';

import ChooseImage from './ChooseImage';

describe('ChooseImage', () => {
  const label = 'label';

  it('renders existing image', () => {
    const value = 'original url';

    render(
      <ChooseImage
        label={label}
        value={value}
      />,
    );

    const imageElement = screen.getByAltText(label);

    expect(imageElement).toHaveAttribute('src', value);
    expect(imageElement.className).toMatchSnapshot();
  });

  it('toggles asset modal', () => {
    const actionText = 'Change';
    const modalTitle = `Choose an image for ${label}`;

    render(<ChooseImage label={label} />);

    const modalButton = screen.getByText(actionText);
    expect(screen.queryByText(modalTitle)).not.toBeInTheDocument();

    modalButton.click();

    screen.getByText(modalTitle);
  });
});
