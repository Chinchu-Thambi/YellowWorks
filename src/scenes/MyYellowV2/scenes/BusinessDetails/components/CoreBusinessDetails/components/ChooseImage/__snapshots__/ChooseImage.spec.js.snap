// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseImage renders existing image 1`] = `"w-full h-full object-cover rounded-md false"`;
